src/dash-table/DataTable.js

```diff
@@ -198,16 +237,20 @@ export const propTypes = {
 
         /**
          * The data-type of the column's data.
-         * `numeric` and `text` are currently the same.
-         * `dropdown` is used to render cells as dropdowns.
+         * 'number': represents both floats and ints
+         * 'text': represents a string
+         * 'any': represents any type of data
+         *
+         * Defaults to 'any' if undefined.
```

Comment:
The columns default to 'any' type if not defined and defaults to a bare boned type configuration (on_change, validation, presentation) if it's not defined, conserving a behaviour that should be consistent with the behaviour before the change